src/d/actor/d_a_tag_kk1.cpp

```diff
@@ -3,64 +3,120 @@
 // Translation Unit: d_a_tag_kk1.cpp
 //
 
+
 #include "d/actor/d_a_tag_kk1.h"
+#include "d/d_com_inf_game.h"
 #include "d/d_procname.h"
 
-/* 000000EC-00000120       .text __ct__15daTag_Kk1_HIO_cFv */
+
+static daTag_Kk1_HIO_c l_HIO;
+
+static const u8 dummy5[] = { 0x00, 0xFF, 0x00, 0x80};
+
 daTag_Kk1_HIO_c::daTag_Kk1_HIO_c() {
-    /* Nonmatching */
+    static f32 a_prm_tbl[] = {350.0f,30.0f,0.0f};
+    mHorizontalDistance = a_prm_tbl[0];
+    mVerticalDistance = a_prm_tbl[1];
+    mUnusedU8 = *reinterpret_cast<u8*>(&a_prm_tbl[2]); //TODO: This u8 appears to be inside the float table. May be a better way to refactor.
```

Comment:
a_prm_tbl seems like it's not an array of floats, but a struct.

If you check the debug map, there's this symbol:
`__as__Q215daTag_Kk1_HIO_c9hio_prm_cFRCQ215daTag_Kk1_HIO_c9hio_prm_c`
Which objdiff's demangler turns into this (a compiler-generated copy assignment function):
`daTag_Kk1_HIO_c::hio_prm_c::operator=(const daTag_Kk1_HIO_c::hio_prm_c&)`

This tells us that inside of the `daTag_Kk1_HIO_c` class body in the header there's struct named `hio_prm_c`. Take a look at d_a_nh for an existing example. These three assignments don't need to be done manually, once you have the struct set up right it will just be:
```suggestion
    static hio_prm_c a_prm_tbl = {
        350.0f,
        30.0f,
        0,
    };
    prm = a_prm_tbl;
```

Reply:
Well spotted. I'll do my best to remember this.